[doc] Link to deprecation policy PEP from the DeprecationWarning documentation (#28123)
python/cpython: Doc/library/exceptions.rst

   (:pep:`565`). Enabling the :ref:`Python Development Mode <devmode>` shows
   this warning.

>    The deprecation policy is described in :pep:`387`.

Reviewer: I'm not sure if "described" is strong enough:

```suggestion
   The deprecation policy is defined by the :pep:`387`.
```
